src/runtime/mini-prompt-api.js

```diff
@@ -0,0 +1,240 @@
+    if (
+      options.autoPromptType !== AutoPromptType.CONTRIBUTION &&
+      options.autoPromptType !== AutoPromptType.SUBSCRIPTION
+    ) {
+      return;
+    }
+
+    const theme = this.clientConfigManager_.getTheme();
+    const lang = this.clientConfigManager_.getLanguage();
+    let textContent = '';
+    if (options.autoPromptType === AutoPromptType.CONTRIBUTION) {
+      textContent = msg(CONTRIBUTION_TITLE_LANG_MAP, lang) || '';
+    } else if (options.autoPromptType === AutoPromptType.SUBSCRIPTION) {
+      textContent = msg(SUBSCRIPTION_TITLE_LANG_MAP, lang) || '';
+    }
+
+    // Create all the elements for the mini prompt.
+    /** @const {!Element} */
+    const miniPromptDiv = createElement(this.doc_.getWin().document, 'div', {
+      'role': 'dialog',
+      'lang': lang,
+    });
+    miniPromptDiv.classList.add(`swg-mini-prompt-${theme}`);
+    const titleContainerDiv = createElement(
+      this.doc_.getWin().document,
+      'div',
+      {'role': 'button'}
+    );
+    titleContainerDiv.classList.add('swg-mini-prompt-title-container');
+    titleContainerDiv./*OK*/ innerHTML = TITLE_CONTAINER_DIV_HTML.replace(
+      '/$theme$/g',
+      theme
+    ).replace('$textContent$', textContent);
+    const closeContainerDiv = createElement(
+      this.doc_.getWin().document,
+      'div',
+      {}
+    );
+    closeContainerDiv.classList.add('swg-mini-prompt-close-button-container');
+    closeContainerDiv./*OK*/ innerHTML = CLOSE_CONTAINER_DIV_HTML.replace(
+      '$theme$',
+      theme
+    );
+    miniPromptDiv.appendChild(titleContainerDiv);
+    miniPromptDiv.appendChild(closeContainerDiv);
+    this.doc_.getWin().document.body.appendChild(miniPromptDiv);
+
+    // Handle events and logging for the various sub-components.
+    const clickFun = () => {
+      this.logClick_(options.autoPromptType);
+      if (typeof options.callback === 'function') {
+        options.callback();
+      }
+      setStyle(miniPromptDiv, 'visibility', 'hidden');
+    };
+    titleContainerDiv.addEventListener('click', clickFun);
+
+    const closeFun = () => {
+      setStyle(miniPromptDiv, 'visibility', 'hidden');
+      this.logClose_(options.autoPromptType);
+    };
+    closeContainerDiv.addEventListener('click', closeFun);
+
+    this.logImpression_(options.autoPromptType);
+  }
+
+  /**
+   * Logs an impression of the mini prompt.
+   * @param {!AutoPromptType} autoPromptType
+   */
+  logImpression_(autoPromptType) {
+    let event;
+    if (autoPromptType === AutoPromptType.CONTRIBUTION) {
+      event = AnalyticsEvent.IMPRESSION_SWG_CONTRIBUTION_MINI_PROMPT;
+    } else if (autoPromptType === AutoPromptType.SUBSCRIPTION) {
+      event = AnalyticsEvent.IMPRESSION_SWG_SUBSCRIPTION_MINI_PROMPT;
+    }
+
+    if (event) {
+      this.eventManager_.logSwgEvent(event, false);
+    }
+  }
+
+  /**
+   * Logs a click of the mini prompt.
+   * @param {!AutoPromptType} autoPromptType
+   */
+  logClick_(autoPromptType) {
```

Comment:
Could you add test coverage for this method?

Reply:
There was test coverage, but there's some weird silent error with the mini-prompt-api-test after I modified the html insertion. Looking into why parts of the test aren't executing now.

Reply:
The tests should `await` the result of `expectMiniPromptCreated()`

Reply:
Fixed.